modin/engines/base/io/text/text_file_reader.py

```diff
@@ -20,11 +21,28 @@
 
 class TextFileReader(FileReader):
     @classmethod
-    def call_deploy(cls, f, chunk_size, num_return_vals, args, quotechar=b'"'):
+    def call_deploy(
```

Comment:
Since we are scanning the file in `call_deploy` for offsets, why not just keep a count of the number of rows here?

Reply:
because it's problematic to determine number of rows for each partition in case we're reading the whole file. We can count the amount of lines in the file in a `for` loop, and then divide that amount on number of splits, but I wonder isn't it will be really slow?

Reply:
ok, found out that it line counting works pretty fast, `0.1212s` for a file with 700000 rows. Is it acceptable time?

Reply:
Since we are already looping over the file and have a chunk of the file contents in a buffer, it will be very fast to use a regular expression to count the number of newlines that do not occur in quotations. Did your evaluation count that separately or reading + line counting takes that amount of time?

Reply:
code that i've measured was something like this
```python
with open(path, "rb") as f:
    lines_counter = 0
    t1 = timer()
    for _ in f:
        lines_counter += 1
    print("time:", timer() - t1)
```

Reply:
It will add even less overhead if you use the regex approach. That is great.

Reply:
Ew, found out that it not so great, file counting lines in a for loop for dataframe with shape `700000, 20` is `0.1212s`, but for `500000, 128` is `2.3168`.
I'll try your approach with regexp and will post time results here

Reply:
Found out that approach with regexp works slower than line counting in a for loop:
```python
with open(path, "rb") as f:
    t1 = timer()
    content = f.read()
    lines_counter = len(re.findall(b"\n", content))
    print("time:", timer() - t1)
```
even that simple regexp works approximately in 1.4 times slower than a for loop above (tested with 500000, 700000, and 5000000 amount of rows). So in the latest commit, I use for loop to count the number of lines in the file

Reply:
to avoid unnecessary overhead of rows counting, I've let `call_deploy` to obtain `nrows` or `chunk_size_bytes` parameter